subprojects/core/src/main/java/org/gradle/api/internal/TaskOutputCaching.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal;
+
+import org.gradle.api.Nullable;
+
+public interface TaskOutputCaching {
```

Comment:
I still have some "off" feeling about the name of this type. This is essentially the `TaskOutputCachingState`, no? Or maybe the `TaskOutputCachingResult`. But I think I prefer state.